crates/forge-runner/src/trace_data.rs

```diff
@@ -0,0 +1,107 @@
+use std::fs;
+use std::path::PathBuf;
+
+// Will be provided by profiler crate in the future
+// This module will be removed!
+use cheatnet::state::CallTrace as InternalCallTrace;
+use serde::{Deserialize, Serialize};
+use starknet_api::core::{ClassHash, ContractAddress, EntryPointSelector};
+use starknet_api::deprecated_contract_class::EntryPointType;
+use starknet_api::transaction::Calldata;
+
+use crate::test_case_summary::{Single, TestCaseSummary};
+
+pub const TRACE_DIR: &str = ".snfoundry_trace";
+
+/// Tree structure representing trace of a call.
+#[derive(Debug, Clone, Serialize)]
+pub struct CallTrace {
+    pub entry_point: CallEntryPoint,
+    pub nested_calls: Vec<CallTrace>,
```

Comment:
It might be memory consuing, later we want to move to id-based graph. There is an issue for it already.

Reply:
Do we need this struct? We could just derive `Serialize` on the `cheatnet::state::CallTrace` 🤔

Reply:
We need this as I simply copied the module to thoth, later we will use it from thoth lib as it will be responsible for standarizing the format.